web/text_layer_builder.js

```diff
@@ -83,13 +91,33 @@ class TextLayerBuilder {
    * @param {number} [timeout] - Wait for a specified amount of milliseconds
    *                             before rendering.
    */
-  render(timeout = 0) {
-    if (!(this.textContent || this.textContentStream) || this.renderingDone) {
+  async render(viewport, timeout = 0) {
+    if (!(this.textContent || this.textContentStream)) {
+      throw new Error(
+        `Neither "textContent" nor "textContentStream" specified.`
+      );
+    }
+
+    const scale = viewport.scale * (globalThis.devicePixelRatio || 1);
+    this.#update(viewport);
+
+    if (this.renderingDone) {
+      this.hide();
+      if (scale !== this.#scale) {
+        updateTextLayer({
+          viewport,
+          scale,
+          textDivs: this.textDivs,
+          textDivProperties: this.textDivProperties,
+          isOffscreenCanvasSupported: this.isOffscreenCanvasSupported,
+        });
+        this.#scale = scale;
+      }
+      this.show();
       return;
     }
-    this.cancel();
 
-    this.textDivs.length = 0;
```

Comment:
Is it actually correct to remove this line, and generally are we cleaning-up properly now, since a *partially rendered* textLayer will have its rendering cancelled if a new `TextLayerBuilder.render`-call happens?
Should we perhaps, in the `cleanup`-method below, add something like the following to prevent any issues in that case?
```javascript
this.textContentItemsStr.length = 0;
this.textDivs.length = 0;
this.textDivProperties = new WeakMap();
```